lib/elastic/transport/client.rb

```diff
@@ -164,14 +165,37 @@ def initialize(arguments = {}, &block)
                          @transport_class.new(hosts: @hosts, options: @arguments)
                        end
         end
+
+        if defined?(::OpenTelemetry) && ENV[OpenTelemetry::ENV_VARIABLE_ENABLED] != 'false'
+          @otel = OpenTelemetry.new
+        end
       end
 
       # Performs a request through delegation to {#transport}.
       #
-      def perform_request(method, path, params = {}, body = nil, headers = nil)
+      def perform_request(method, path, params = {}, body = nil, headers = nil, opts = {})
         method = @send_get_body_as if 'GET' == method && body
         validate_ca_fingerprints if @ca_fingerprint
-        transport.perform_request(method, path, params, body, headers)
+        if @otel
+          span_name = opts[:endpoint] || method
+          @otel.tracer.in_span(span_name) do |span|
+            span['http.request.method'] = method
+            opts[:defined_params]&.each do |k, v|
+              if v.respond_to?(:join)
+                span["db.elasticsearch.path_parts.#{k}"] = v.join(',')
+              else
+                span["db.elasticsearch.path_parts.#{k}"] = v
+              end
+            end
+            if body_as_json = @otel.process_body(body, opts[:endpoint])
+              span['db.statement'] = body_as_json
+            end
+            span['db.operation'] = opts[:endpoint] if opts[:endpoint]
+            transport.perform_request(method, path, params || {}, body, headers)
```

Comment:
Since `params` is being passed with the default value of `{}` in the `perform_request` signature, I think we don't need `|| {}` here and [on line 197](https://github.com/elastic/elastic-transport-ruby/pull/54/files#diff-78264cee2af175e6fd5eed7e26abeade1785739ab1d3ac97cf41f26ac128361bR197), right?.

Reply:
With my testing, there was sometimes a case when there were no params but there was a body so the `perform_request` method would be called with the params arg explicitly set to `nil`. So once it gets to [this line](https://github.com/elastic/elastic-transport-ruby/blob/main/lib/elastic/transport/transport/base.rb#L283) you get an `NoMethodError: undefined method `delete' for nil:NilClass` error.

I think this scenario is already possible before these changes:
```ruby
$ client.perform_request('POST', '/', nil, '{"foo":"bar"}', '{"Content-Type":"application/x-ndjson"}')
NoMethodError: undefined method `delete' for nil:NilClass
from ..../elastic-transport-ruby/fork/elastic-transport-ruby/lib/elastic/transport/transport/base.rb:283:in `perform_request'
```